sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1516,3 +1517,48 @@ def test_multi_core_gridsearch_and_early_stopping():
                                 random_state=0)
     search.fit(iris.data, iris.target)
     assert search.best_score_ > 0.8
+
+
+@pytest.mark.skipif(
+        not hasattr(sp, 'random'),
+        reason="this test uses scipy.random, that was introduced in version  "
+        "0.17. This skip condition can be dropped as soon as we drop support "
+        "for scipy versions older than 0.17")
+def test_SGDClassifier_fit_for_all_backends():
+    # This is a non-regression smoke test. In the multi-class case,
+    # SGDClassifier.fit fits each class in a one-versus-all fashion using
+    # joblib.Parallel.  However, each OvA step updates the coef_ attribute of
+    # the estimator in-place. Internally, SGDClassifier calls Parallel using
+    # require='sharedmem'. This test makes sure SGDClassifier.fit works
+    # consistently even when the user asks for a backend that does not provide
+    # sharedmem semantics.
+
+    # We further test a case where memmapping would have been used if
+    # SGDClassifier.fit was called from a loky or multiprocessing backend. In
+    # this specific case, in-place modification of clf.coef_ would have caused
+    # a segmentation fault when trying to write in a readonly memory mapped
+    # buffer.
+
+    random_state = 42
+
+    # Create a classification problem with 50000 features and 20 classes. Using
+    # loky or multiprocessing this make the clf.coef_ exceed the threshold
+    # above which memmaping is used in joblib and loky (1MB as of 2018/11/1).
+    X = sp.random(1000, 50000, density=0.01, format='csr',
+                  random_state=random_state)
+    y = np.random.choice(20, 1000)
+
+    # begin by fitting a SGD classifier sequentially
+    clf = SGDClassifier(tol=1e-3, max_iter=1000, n_jobs=1,
+                        random_state=random_state)
+    clf.fit(X, y)
+    coef_sequential_fitting = clf.coef_
+
+    # fit a SGDClassifier for each available backend, and make sure the
+    # coefficients are equal to those obtained using a sequential fit
+    for backend in ['threading', 'multiprocessing', 'loky']:
+        clf = SGDClassifier(tol=1e-3, max_iter=1000, n_jobs=4,
```

Comment:
You can parametrize this.

Reply:
What do you mean?

Reply:
instead of making a loop inside the test, you can do
```python
@pytest.mark.parametrize("backend", ["threading", "multiprocessing", "loky"])
def test_SGDClassifier_fit_for_all_backends(backend):
   do something with backend
```
And pytest will run the test for all values of backend

Reply:
Did not know about that, thanks for explaining!